apollo-router/src/axum_factory/axum_http_server_factory.rs

```diff
@@ -644,10 +647,7 @@ async fn handle_graphql(
             .in_current_span();
         let res = match tokio::task::spawn(task).await {
             Ok(res) => res,
-            Err(err) => {
-                let msg = format!("router service call failed: {err}");
-                return (StatusCode::INTERNAL_SERVER_ERROR, msg).into_response();
-            }
+            Err(err) => return internal_server_error(err),
```

Comment:
I'm just wondering if we could have that kind of critical errors coming before handle_graphql, if so what's the behavior right now ? Do we also want to change this ? And if it's an error coming from a tower layer or so ?

Reply:
I think we should open some followups to investigate this and also deal with other non-graphql errors that are escaping in handle_graphql such as rate limiting.

Reply:
Can we do that before/as we land?

Reply:
Added https://github.com/apollographql/router/issues/5171